### PR TITLE
refactor quoting logic into QuotedMessagePresenter

### PR DESCRIPTION
**Pull request by Valodim**
what the title says.

this PR should introduce no semantic changes. MessageCompose received a whole -940 lines, and is down to four warnings now, woop woop.

this is in preparation for crypted replies - QuotedMessagePresenter will probably have to learn how to decrypt messages somewhere down the road.

### CODE REVIEW

File: k9mail/src/main/java/com/fsck/k9/ui/compose/QuotedMessagePresenter.java [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+     * <p><strong>Note:</strong>
+     * When this field is not {@code null} we assume that the message we are composing right now
+     * should be encrypted.
+     * </p>
+     */
+    private String sourceMessageBody;
+
+    private SimpleMessageFormat quotedTextFormat;
+    private InsertableHtmlContent quotedHtmlContent;
+    private Account account;
```
Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/ui/compose/QuotedMessagePresenter.java]: note to self: this variable isn't updated properly on account change

Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/ui/compose/QuotedMessagePresenter.java]: (done)

**Comment by Valodim** [comment]
I would appreciate a quick review for this one since the decrypted reply branch currently depends on two other development branches.

**Comment by cketti** [comment]
Looks like the presenter still contains a lot of logic that could be extracted. And now that the classes are smaller and better separated it would be nice to have tests. But I guess it's still better than what we had before :)

**Pull request merged**